src/pages/suggestions/[suggestion].tsx

```diff
@@ -0,0 +1,138 @@
+import { FC, useState } from "react"
+import { MusicalNoteIcon, XMarkIcon } from "@heroicons/react/24/solid"
+import Link from "next/link"
+import ProgressionBar from "@/components/suggestion/progression-bar"
+import Image from "next/image"
+import { Instruments } from "@/constants/instruments"
+import { ISuggestion } from "@/interfaces/suggestion"
+import WithProtectedRoute from "@/components/protected-route/protected-route"
+import { GetStaticPaths, GetStaticProps } from "next"
+import { getSuggestion, updateSuggestion } from "@/services/suggestion.service"
+import { useAuthContext } from "@/context/auth-context"
+
+interface SuggestionProps {
+  props: ISuggestion
+}
+
+const Suggestion: FC<SuggestionProps> = ({ props }) => {
+  const [suggestion, setSuggestion] = useState<ISuggestion>(props)
+  const { user } = useAuthContext()
+  const username = user?.additionalUserData.username as string
+
+  const selectInstrument = (index: number) => {
+    if (suggestion.roles.at(index)?.filledBy?.includes(username)) {
```

Comment:
I think using a reference path to store the user and role is more desirable for in the future since you won't have to deal with usenames changing for example & the hassle of dealing with disabled & removed users